uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -227,14 +227,12 @@ define([
         },
 
         /**
-         * Delete a leave request
+         * Deletes the leave request
          *
-         * @return {Promise} Resolved with {Object} Deleted Leave request
+         * @return {Promise}
          */
         delete: function () {
-          this.is_deleted = true;
-
-          return LeaveRequestAPI.update(this.toAPI());
+          return LeaveRequestAPI.delete(this.id);
```

Comment:
As discussed with @davialexandre , the `is_deleted` flag will not be exposed to the frontend, being an internal backend implementation (the backend automatically filters the soft-deleted leave requests when returning their list, for example).

So basically the frontend will not know that there is a soft-delete at all, it will just think that the requests get deleted, period.

So in the future (I can't see any current usage of the `is_deleted` flag in the whole L&A app), the angular app will simply remove the deleted request from the list, i guess via emitting an event, like
```js
// deleting the request
.then(function () {
  $rootScope.$emit('LeaveRequest::deleted', deletedRequest)
});